### PR TITLE
got subsurfaces to display

### PR DESCRIPTION
**Pull request by wmww**

### CODE REVIEW

File: src/server/frontend_wayland/wl_mir_window.cpp [hunk 1]
```diff
@@ -121,7 +135,9 @@ void WlAbstractMirWindow::commit()
     if (params->size == geometry::Size{})
         params->size = geometry::Size{640, 480};
 
-    params->streams = std::move(std::vector<shell::StreamSpecification>{{mir_surface->stream_id, mir_surface->buffer_offset, {}}});
+    params->streams = std::vector<shell::StreamSpecification>();
```
Comment (AlanGriffiths) [inline src/server/frontend_wayland/wl_mir_window.cpp]: We could use uniform initialization syntax. Vis: `params->streams = std::vector<shell::StreamSpecification>{};`

File: src/server/frontend_wayland/wl_mir_window.h [hunk 2]
```diff
@@ -58,10 +59,12 @@ class WlAbstractMirWindow : public WlMirWindow
 {
 public:
     WlAbstractMirWindow(wl_client* client, wl_resource* surface, wl_resource* event_sink,
-        std::shared_ptr<frontend::Shell> const& shell);
+                        std::shared_ptr<frontend::Shell> const& shell);
```
Comment (AlanGriffiths) [inline src/server/frontend_wayland/wl_mir_window.h]: Random whitespace change

File: src/server/frontend_wayland/wl_subcompositor.cpp [hunk 3]
```diff
@@ -90,8 +93,14 @@ void mf::WlSubsurface::new_buffer_size(geometry::Size const& buffer_size)
...
 
+void mf::WlSubsurface::invalidate_buffer_list()
+{
+    parent->role->invalidate_buffer_list();
+}
+
 void mf::WlSubsurface::commit()
 {
+    invalidate_buffer_list();
     // TODO
```
Comment (AlanGriffiths) [inline src/server/frontend_wayland/wl_subcompositor.cpp]: Do we still need a TODO here?

File: src/server/frontend_wayland/wl_mir_window.h (hunk 2 above)
Comment (wmww) [reply src/server/frontend_wayland/wl_mir_window.h]: https://mir-server.io/doc/cppguide/index.html?showone=Function_Declarations_and_Definitions#Function_Declarations_and_Definitions

**Pull request merged**